src/compiler/utilities.ts

```diff
@@ -1159,51 +1159,51 @@ namespace ts {
         return (<CallExpression | Decorator>node).expression;
     }
 
-    export function nodeCanBeDecorated(node: Node): boolean {
+    export function nodeCanBeDecorated(node: Node, parent: Node = node.parent, grandparent: Node = parent.parent): boolean {

```

Comment:
Rather than add `parent` and `grandparent` as optional arguments, I'd rather make them required. We can just modify the checker to pass in `node.parent` and `node.parent.parent` to the call. That way the function won't be possibly misused in other places later.